adatest/generators.py

```diff
@@ -226,6 +232,45 @@ def prepare_inputs_for_generation(input_ids, **kwargs):
         return self._parse_suggestion_texts(suggestion_texts, prompts)
 
 
+class Pipelines(HuggingFace):
+    import transformers
+    def __init__(self, pipeline: transformers.pipelines.base.Pipeline , sep="\n", subsep=" ", quote="\"", filter=profanity.censor):
+        super().__init__(pipeline, sep, subsep, quote, filter)
+        self.gen_type = "model"
+        self.stop_sequence = self.quote + self.sep
+        self._sep_stopper = HuggingFace.StopAtSequence(self.stop_sequence, pipeline.tokenizer)
+
+    def __call__(self, prompts, topic, topic_description, mode, scorer=None, num_samples=1, max_length=100):
+        if len(prompts) == 0:
+            raise ValueError("ValueError: Unable to generate suggestions from completely empty TestTree. Consider writing a few manual tests before generating suggestions.") 
+        prompts, prompt_ids = self._validate_prompts(prompts)
+        prompt_strings = self._create_prompt_strings(prompts, topic, mode)
+
+        suggestion_texts = []
+        for p in prompt_strings:
+            prompt_length = len(self.source.tokenizer.tokenize(p))
+            self._sep_stopper.prompt_length = prompt_length
+            self._sep_stopper.max_length = max_length
+            generations = self.source(p,
+                        do_sample=True,
+                        max_length=prompt_length + max_length,
+                        num_return_sequences=num_samples,
+                        pad_token_id=self.source.model.config.eos_token_id,
+                        stopping_criteria=[self._sep_stopper])
+            for gen in generations:
+                generated_text: str = gen['generated_text'][len(p):]
```

Comment:
I think we can skip this type annotation on this inner variable, right? Unless it's a particularly tricky output

Reply:
Removed.